src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -1305,27 +1305,27 @@ internal void UpdateResourceBindings(ResourceUpdateReason updateReason, Resource
 
 			if (updateReason == ResourceUpdateReason.ThemeResource)
 			{
-				dictionariesInScope = GetResourceDictionaries(includeAppResources: false, containingDictionary).ToArray();
+				dictionariesInScope = GetResourceDictionaries(includeAppResources: false, resourceContextProvider, containingDictionary).ToArray();
 				for (var i = dictionariesInScope.Length - 1; i >= 0; i--)
 				{
 					ResourceResolver.PushSourceToScope(dictionariesInScope[i]);
 				}
 
 				_properties.UpdateBindingExpressions();
 
-				foreach (var dict in dictionariesInScope)
+				for (int i = 0; i < dictionariesInScope.Length; i++)
 				{
 					ResourceResolver.PopSourceFromScope();
 				}
 			}
 
 			if (_resourceBindings == null || !_resourceBindings.HasBindings)
 			{
-				UpdateChildResourceBindings(updateReason);
+				UpdateChildResourceBindings(updateReason, ActualInstance as FrameworkElement ?? resourceContextProvider);
```

Comment:
we could skip the xaml generator part, and solely rely on this to inject the parent context.
either way this is needed, because otherwise, it would override previously correct resolution from xaml-generated code

the question is do we want to keep the source-generator part of this pr?
@jeromelaban, CC: @Youssef1313

Reply:
xaml generator part dropped;
just in case, for curiosity, the old change set are here: https://github.com/unoplatform/uno/tree/dev/xygu/20240618/resource-resolution-full https://github.com/unoplatform/uno/commit/99c5c24af8a94b29e0b9b3dace5716ea44b03f98